### ISSUE TITLE
Truffle debug : TypeError: Cannot read property 'children' of undefined

### ISSUE THREAD

**Issue by carbonnetwork** [trufflesuite/truffle#1086]
```
$ truffle debug 0xf174b473a1e7f0a3e7911b90821fe4427f2bd88da9b51fbb3f1017024115e5e4
Compiling ./contracts/AdminController.sol...
Compiling ./contracts/AnalysisController.sol...
Compiling ./contracts/Datastore.sol...
Compiling ./contracts/DemandController.sol...
Compiling ./contracts/Finance.sol...
Compiling ./contracts/Lockstore.sol...
Compiling ./contracts/Manager.sol...
Compiling ./contracts/Migrations.sol...
Compiling ./contracts/lib/ArraysLib.sol...
Compiling ./contracts/lib/MapLib.sol...
Compiling ./contracts/lib/StringLib.sol...
Compiling ./contracts/test/Array.sol...
Compiling carbon-ico/contracts/Ownable.sol...
Compiling carbon-ico/contracts/SafeMathLib.sol...

Gathering transaction data...

Addresses affected:
 0xb1efdc37a0db4d78e51a43ff6d03ebda6d1b31d8 - MapLib
 0x42f36dce670e84e846b3b9652d8923fdfb825ee2 - Lockstore

Commands:
(enter) last command entered (step next)
(o) step over, (i) step into, (u) step out, (n) step next
(;) step instruction, (p) print instruction, (h) print this help, (q) quit
(b) toggle breakpoint, (c) continue until breakpoint
(+) add watch expression (`+:<expr>`), (-) remove watch expression (-:<expr>)
(?) list existing watch expressions
(v) print variables and values, (:) evaluate expression - see `v`


Lockstore.sol:

11:  * The Lock contract does this and that...
12:  */
13: contract Lockstore is Manager {
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

debug(development:0xf174b473...)> +:keyIndex
undefined
debug(development:0xf174b473...)>

Lockstore.sol:
.........
.......
........
:keyIndex
  TypeError: Cannot read property 'children' of undefined
    at printWatchExpressionResult (/usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-core/lib/commands/debug.js:214:1)
    at evalAndPrintExpression (/usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-core/lib/commands/debug.js:269:1)
    at Session.view (/usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-debugger/dist/debugger.js:2236:1)
    at /usr/local/lib/node_modules/truffle/build/webpack:/~/reselect/lib/index.js:36:1
    at /usr/local/lib/node_modules/truffle/build/webpack:/~/reselect/lib/index.js:86:1
    at args (/usr/local/lib/node_modules/truffle/build/webpack:/~/reselect-tree/index.js:209:1)
    at Function.<anonymous> (/usr/local/lib/node_modules/truffle/build/webpack:/~/reselect/lib/index.js:36:1)
    at /usr/local/lib/node_modules/truffle/build/webpack:/~/reselect/lib/index.js:90:1
    at /usr/local/lib/node_modules/truffle/build/webpack:/~/reselect/lib/index.js:36:1
    at /usr/local/lib/node_modules/truffle/build/webpack:/~/reselect/lib/index.js:76:1
    at data.current.identifiers.decoded (/usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-debugger/dist/debugger.js:1689:148)
    at Array.map (<anonymous>)
    at map (/usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-debugger/dist/debugger.js:1690:1)
    at /usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-debugger/dist/debugger.js:1592:1
    at decode (/usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-debugger/dist/debugger.js:3847:1)
    at decodeStorageReference (/usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-debugger/dist/debugger.js:3828:38)
```
If I remove the watch expression "keyIndex", It has no this exception.

```
$ truffle version
Truffle v4.1.13 (core: 4.1.13)
Solidity v0.4.24 (solc-js)
```

I have no idea about this. Could some one help me. Thanks.

**Comment by LoomiDigital** [comment]
@cgewecke Hey, I'm also getting this issue when attempting to watch certain variables. Not sure if it's due to the fact that some of the calculations are done from values inherited from another contract.

Do you happen to have any insights into the issue? As it stands I'm finding the debugger unusable.

**Comment by cgewecke** [comment]
@LoomiDigital Very sorry for the delay responding.  I think you might be right about storage from inherited contracts being a problem and I believe it's on the roadmap for debugger improvements. 

@gnidan might have more insight this.

**Comment by vindberg** [comment]
This also seem to be the case when using a single .sol file.

**Comment by gnidan** [comment]
@haltman-at says: if this is related to inheritance, it should be fixed in #1573, otherwise we will have to look into it.

In the meantime, if you get a chance, could you try this on a later version of Truffle and let us know if it's still a problem for you? Thanks!

**Comment by CruzMolina** [comment]
Closing for issue maintenance. Happy to reopen if this is still an issue!
